server/src/main/scala/org/labrad/browser/Controllers.scala

```diff
@@ -1,21 +0,0 @@
-package org.labrad.browser
-
-import play.api.libs.concurrent.Execution.Implicits.defaultContext
-import play.api.libs.json._
-import play.api.mvc._
-import scala.concurrent.Future
-
-trait JsonRpc { self: Controller =>
-
-  def rpc[A: Reads, B: Writes](f: A => Future[B]) = Action.async(BodyParsers.parse.json) { request =>
-    val originOpt = request.headers.get("Origin")
-    val a = request.body.as[A]
-    f(a).map { b =>
-      val headers = Seq.newBuilder[(String, String)]
-      for (origin <- originOpt) {
-        headers += "Access-Control-Allow-Origin" -> origin
-      }
-      Ok(Json.toJson(b)).withHeaders(headers.result: _*)
-    }
-  }
-}
```

Comment:
This was just a little utility for creating actions that handle single http requests, taking a json object as the request body and returning a json object as the response body. Since the api calls now all use real jsonrpc over websockets, this is no longer needed. "Controller" is a play framework thing, and we're now only using it for a few things (serving the index.html page and other static files, and handling the websocket connection request). Most of those other files which implement apis are no longer "controllers" in the sense of the play framework, so those files could be renamed.